app/gui2/src/components/GraphEditor/GraphEdge.vue

```diff
@@ -15,7 +16,8 @@ const props = defineProps<{
   edge: Edge
   nodeRects: Map<ExprId, Rect>
   exprRects: Map<ExprId, Rect>
-  exprNodes: Map<ExprId, ExprId>
+  lookupExprNode: (exprId: ExprId) => ExprId | undefined
+  lookupNodeColor: (nodeId: ExprId) => string | undefined
```

Comment:
I tried to avoid using the graph store directly in the edge, but there probably is no point. We are already referencing the selection there anyway. So I'll just simplify that.